### ISSUE TITLE
Nothing appears on the CRT screen - scrambled characters in Settings GUI

### ISSUE THREAD

**Issue by rantonels** [Swordfish90/cool-retro-term#217]
Program starts up without crashing, but I am greeted by a black window with a scanline effect. No characters appear, and it seems to be pretty much unresponsive to keyboard input. Opening the setting panels reveals most words in the GUI are somewhat  scrambled, with missing and misplaced characters.
![2015-01-28-135820_1024x768_scrot](https://cloud.githubusercontent.com/assets/5089595/5937272/c9b19b3c-a6ed-11e4-89ca-d5ed7dac0592.png)
![2015-01-28-135926_580x400_scrot](https://cloud.githubusercontent.com/assets/5089595/5937271/c9b0ff42-a6ed-11e4-9867-169ea27e83eb.png)

When running, output includes the cryptic line:

```
QObject::connect: No such slot Konsole::TerminalDisplay_QML_42::close()
```

I am using Debian Jessie (Crunchbang) on an HP Pavilion dv7. I'm running the program on the main Intel GPU, not the NVIDIA.

project was downloaded with ```git --recursive``` and compiled from source without any errors.

Full output from ```./cool-retro-term``` is:

```
Fontconfig warning: "/etc/fonts/conf.d/65-droid-sans-fonts.conf", line 103: Having multiple values in <test> isn't supported and may not work as expected
Fontconfig warning: "/etc/fonts/conf.d/65-droid-sans-fonts.conf", line 138: Having multiple values in <test> isn't supported and may not work as expected
default KB_LAYOUT_DIR:  "/home/riccardo/cool-retro-term/qmltermwidget/QMLTermWidget/kb-layouts"
QObject::connect: No such slot Konsole::TerminalDisplay_QML_42::close()
loadAllColorSchemes
Session::run() - program: "/bin/bash"
Session::run() - arguments: ("")
started!
Fontconfig warning: "/etc/fonts/conf.d/65-droid-sans-fonts.conf", line 103: Having multiple values in <test> isn't supported and may not work as expected
Fontconfig warning: "/etc/fonts/conf.d/65-droid-sans-fonts.conf", line 138: Having multiple values in <test> isn't supported and may not work as expected
```

**Comment by Swordfish90** [comment]
Hi @rantonels, this is extremely strange, and my first guess is that you have issues with fontconfig on your system. The "No such slot error" is harmless and not the cause of this (although I should fix it). I suggest you to try to fix your fontconfig settings or file a bug for your distribution. Just ping me if you need a minimal QML to reproduce the issue (just a window with some labels should suffice).

**Comment by rantonels** [comment]
A very small update: 

upgrading package "fonts-droid" has replace the conf file /etc/fonts/conf.avail/65-droid-sans-fonts.conf and fixed the Fontconfig warning. This of course should have nothing to do with cool-retro-term and in fact nothing changed on that front.

fontconfig itself is at the latest version.

Thanks, I would love some minimal code to try and identify where the problem lies.

**Comment by Swordfish90** [comment]
Ok. If the fontconfig warning was a false alarm I suggest you to try some of the default QML files from digia, such as this one: http://doc.qt.io/qt-5/qtquickcontrols-basiclayouts-main-qml.html . Copy the content into an empty text file, save it as main.qml and run it with "qmlscene main.qml" . If you see any missing glyphs file a bug for your distro.

Something else you might want to try is to use the CPU to draw the terminal texture (not all the effects which would made it basically useless). To do that comment the following line in qmltermwidget/lib/TerminalDisplay.cpp

	setRenderTarget(QQuickPaintedItem::FramebufferObject);

Let me know if anything changes.

**Comment by rantonels** [comment]
I have tested the main.qml file you linked, it works correctly, if not for some weird flicker when resizing the window. Text labels are correct.

I have tried recompiling with the

```
setRenderTarget(QQuickPaintedItem::FramebufferObject);
```

line commented out, and it compiles correctly. However, when running I get the same behaviour as before.

**Comment by Swordfish90** [comment]
Ok, this is getting extremely strange. Please try those additional tests:

1) Run everything is software mode (this will be slow as hell)
LIBGL_ALWAYS_SOFTWARE=1 cool-retro-term
2) Post the output of this: QSG_INFO=1 cool-retro-term
3) Open the settings window with qmlscene: qmlscene app/qml/SettingsWindow.qml (This will give you plenty of errors, but are all the characters visible?).

**Comment by rantonels** [comment]
```
LIBGL_ALWAYS_SOFTWARE=1 ./cool-retro-term
```

gives... this:

![2015-01-30-000002_1024x768_scrot](https://cloud.githubusercontent.com/assets/5089595/5967206/0a639ac2-a80b-11e4-9fe0-43d7b6f32fd0.png)

it... moves.

```
QSG_INFO=1 cool-retro-term
```

yields:

```
QSG: basic render loop
default KB_LAYOUT_DIR:  "/home/riccardo/cool-retro-term/qmltermwidget/QMLTermWidget/kb-layouts"
QObject::connect: No such slot Konsole::TerminalDisplay_QML_54::close()
loadAllColorSchemes
Session::run() - program: "/bin/bash"
Session::run() - arguments: ("")
started!
QSG: texture atlas dimensions: 1024 x 1024
R/G/B/A Buffers:    8 8 8 8
Depth Buffer:       24
Stencil Buffer:     8
Samples:            -1
GL_VENDOR:          Tungsten Graphics, Inc
GL_RENDERER:        Mesa DRI Intel(R) Ivybridge Mobile 
GL_VERSION:         3.0 Mesa 8.0.5
GL_EXTENSIONS:      GL_EXT_provoking_vertex GL_ARB_point_parameters GL_EXT_secondary_color GL_APPLE_vertex_array_object GL_EXT_texture_swizzle GL_EXT_copy_texture GL_EXT_subtexture GL_MESA_window_pos GL_ARB_vertex_shader GL_NV_texture_rectangle GL_ATI_envmap_bumpmap GL_EXT_blend_subtract GL_INGR_blend_func_separate GL_ARB_fragment_coord_conventions GL_SGIS_generate_mipmap GL_NV_depth_clamp GL_EXT_texture_env_dot3 GL_ARB_shader_objects GL_IBM_multimode_draw_arrays GL_EXT_texture_snorm GL_NV_packed_depth_stencil GL_EXT_stencil_wrap GL_MESA_texture_array GL_ARB_texture_swizzle GL_EXT_texture_lod_bias GL_ARB_half_float_pixel GL_ARB_draw_elements_base_vertex GL_ARB_depth_texture GL_EXT_point_parameters GL_EXT_texture_env_combine GL_ARB_depth_clamp GL_ARB_robustness GL_EXT_packed_pixels GL_ARB_framebuffer_sRGB GL_EXT_blend_equation_separate GL_EXT_polygon_offset GL_EXT_fog_coord GL_ARB_vertex_array_object GL_IBM_rasterpos_clip GL_EXT_texture_filter_anisotropic GL_EXT_texture_sRGB_decode GL_ARB_ES2_compatibility GL_ATI_separate_stencil GL_EXT_framebuffer_blit GL_EXT_blend_color GL_MESA_texture_signed_rgba GL_EXT_texture_sRGB GL_EXT_texture3D GL_EXT_rescale_normal GL_ARB_texture_compression GL_ARB_depth_buffer_float GL_EXT_texture_object GL_EXT_texture_edge_clamp GL_NV_vertex_program1_1 GL_ARB_explicit_attrib_location GL_SGIS_texture_lod GL_EXT_framebuffer_sRGB GL_OES_read_format GL_ATI_draw_buffers GL_ATI_blend_equation_separate GL_ARB_copy_buffer GL_NV_texgen_reflection GL_EXT_framebuffer_multisample GL_ARB_map_buffer_range GL_EXT_separate_shader_objects GL_EXT_packed_depth_stencil GL_APPLE_object_purgeable GL_EXT_framebuffer_object GL_ARB_texture_border_clamp GL_ARB_texture_env_crossbar GL_ARB_texture_compression_rgtc GL_EXT_draw_range_elements GL_EXT_blend_func_separate GL_EXT_texture_compression_rgtc GL_NV_vertex_program GL_ARB_sync GL_ARB_texture_mirrored_repeat GL_SGIS_texture_border_clamp GL_EXT_texture_env_add GL_ARB_sampler_objects GL_ARB_seamless_cube_map GL_EXT_texture_array GL_EXT_bgra GL_ARB_point_sprite GL_ARB_framebuffer_object GL_3DFX_texture_compression_FXT1 GL_ARB_texture_rectangle GL_ARB_fragment_program_shadow GL_EXT_texture_integer GL_ARB_fragment_shader GL_EXT_texture_shared_exponent GL_ARB_vertex_array_bgra GL_ARB_occlusion_query GL_ARB_texture_non_power_of_two GL_ARB_shadow GL_EXT_multi_draw_arrays GL_ARB_texture_env_add GL_ATI_texture_env_combine3 GL_EXT_timer_query GL_ATI_texture_float GL_ARB_fragment_program GL_ARB_vertex_program GL_EXT_shadow_funcs GL_MESA_ycbcr_texture GL_OES_EGL_image GL_EXT_transform_feedback GL_IBM_texture_mirrored_repeat GL_EXT_stencil_two_side GL_MESA_pack_invert GL_ARB_pixel_buffer_object GL_EXT_abgr GL_EXT_separate_specular_color GL_EXT_texture GL_NV_texture_env_combine4 GL_EXT_packed_float  GL_EXT_vertex_array GL_ARB_multisample GL_EXT_texture_rectangle GL_ARB_texture_cube_map GL_SUN_multi_draw_arrays GL_EXT_vertex_array_bgra GL_EXT_gpu_program_parameters GL_ARB_texture_rg GL_EXT_blend_minmax GL_ARB_transpose_matrix GL_ARB_color_buffer_float GL_EXT_compiled_vertex_array GL_ARB_vertex_buffer_object GL_ARB_provoking_vertex GL_ARB_shading_language_100 GL_ARB_half_float_vertex GL_NV_light_max_exponent GL_NV_conditional_render GL_ARB_texture_float GL_APPLE_packed_pixels GL_SGIS_texture_edge_clamp GL_ARB_texture_env_dot3 GL_NV_blend_square GL_ARB_texture_env_combine GL_ARB_multitexture GL_EXT_texture_cube_map GL_ARB_shader_texture_lod GL_ARB_window_pos GL_EXT_draw_buffers2 GL_ARB_draw_buffers GL_EXT_pixel_buffer_object
```

now 

```
qmlscene app/qml/SettingsWindow.qml 
```

gives this:

![2015-01-30-000222_580x400_scrot](https://cloud.githubusercontent.com/assets/5089595/5967226/45b7cf9e-a80b-11e4-9d20-aa2131d4d517.png)

![2015-01-30-000708_580x400_scrot](https://cloud.githubusercontent.com/assets/5089595/5967250/9828bcde-a80b-11e4-9de6-496ede5a01cf.png)


which is **not** ok, looking around tabs stuff is misplaced or missing, but it seems better than before.

**Comment by rantonels** [comment]
All issues were resolved by a libmesa and Linux kernel upgrade (3.2 -> 3.16). No recompilation was necessary.

**Comment by Swordfish90** [comment]
I'm sorry, I forgot to answer here. Anyway, I'm glad you resolved your problems, it seems it was a driver issues after all.
